### PR TITLE
Support snippets in autocomplete suggestions

### PR DESCRIPTION
**Pull request by hansec**
Add support for snippets in completion items returned by language servers.

### CODE REVIEW

File: lib/adapters/autocomplete-adapter.js [hunk 1]
```diff
@@ -101,6 +101,10 @@ export default class AutocompleteAdapter {
       description: item.documentation,
       descriptionMarkdown: item.documentation,
     };
+    if (item.insertTextFormat == 2) {
+      sugg.snippet = item.insertText;
+    };
+    return sugg;
```
Comment (damieng) [inline lib/adapters/autocomplete-adapter.js]: Thanks for this! Could we maybe go with just a 1-line change version of having the snippet added, e.g.

snippet: item.insertTextFormat == 2 ? item.insertText : undefined

**Comment by damieng** [comment]
Fixes #31

**Comment by hansec** [comment]
Simplified as requested. Sorry for the verbosity, Javascript is not my primary language.

File: lib/adapters/autocomplete-adapter.js [hunk 2]
```diff
@@ -96,6 +96,7 @@ export default class AutocompleteAdapter {
       text: item.insertText || item.label,
       displayText: item.label,
       filterText: item.filterText || item.label,
+      snippet: item.insertTextFormat == 2 ? item.insertText : undefined,
```
Comment (damieng) [inline lib/adapters/autocomplete-adapter.js]: Sorry, my mistake. It's supposed to be `===` to keep the linter happy.  (JS not the language I'm most familiar with either ;-)  )

**Pull request merged**